android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCard.kt

```diff
@@ -0,0 +1,53 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.mock.MoviesMock
+import agency.five.codebase.android.movieapp.model.Movie
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+
+
+@Composable
+fun MovieCard(
+    modifier: Modifier = Modifier,
+    movie: Movie,
```

Comment:
Instead of passing domain object we should pass custom object : `data class MovieCardViewState `
which has only necessary properties. This was component can be reused. If we pass domain object it is coupled to it and we want to avoid that. 
```
data class MovieCardViewState(
    val imageUrl: String,
    val isFavorite: Boolean,
)
```